### DISCUSSION TITLE
How to use less memory to avoid OOM in ios app webview?

### DISCUSSION THREAD

**Discussion by poi-mashiro** [fabricjs/fabric.js#9950] [General]
I use fabric v5 to implement an image editor on PC browsers. I want to reuse same code in mobile app. But some iphone's physical memory may be too small. I get white/black screen when I reuse the code only for rendering result. 

Is there some fabric config to improve it easily?

**Comment by asturur** [comment]
You can disable caching as a first attempt, but iphone are notorious to have very few memory allowed for canvas applications in their browser.
Also avoid super large canvases, use canvas that are at max the css size of the screen.

**Comment by mhieupham1** [comment]
- first, please set objectCaching = false for all objects, except for the pen objects. I couldn't find a way to clear the canvas cache with the eraser.
- second, you need to override the createCanvasElement function to save all the canvases that are created, and you need to delete them by setting the width and height to 1. I was able to use a canvas with a size of 1600 x 900 with 1000 objects including pen, textbox, shape, and image without encountering memory errors on iPad OS 7th. @poi-mashiro

**Comment by poi-mashiro** [comment]
I found out that it was caused by the woff file. In my scenario, I sometimes had to load a total of 30MB of woff files.

In Iphone X with 3G RAM,  "https://benchmarks.slaylines.io/fabric.html" can run with 64000.

**Reply by poi-mashiro** [reply]
Adobe's Source Han Sans has some error in woff. Or it has too many characters

**Reply by poi-mashiro** [reply]
ios 16.1.1 has some error in woff.  Ipad 6th with 2G RAM, ios 14 can run and not OOM.
Update ios 16.1.1 to 16.7.8 and not OOM.
